contracts/tasks/aero.js

```diff
@@ -0,0 +1,162 @@
+const { formatUnits } = require("ethers").utils;
+const { BigNumber } = require("ethers");
+const { getBlock } = require("./block");
+const { resolveAsset, resolveContract } = require("../utils/resolvers");
+const { getSigner } = require("../utils/signers");
+const { base } = require("../utils/addresses");
+
+const snapAero = async ({ block }) => {
+  const signer = await getSigner();
+  const blockTag = await getBlock(block);
+
+  const weth = await resolveAsset("WETH");
+  const vault = await resolveContract("OETHBaseVaultProxy", "IVault");
+  const oeth = await resolveContract("OETHBaseProxy", "OETHBase");
+  const pool = await resolveContract(base.aerodromeOETHbWETHClPool, "ICLPool");
+  const aeroStrat = await resolveContract(
+    `AerodromeAMOStrategyProxy`,
+    "AerodromeAMOStrategy"
+  );
+  const sugarHelper = await resolveContract(base.sugarHelper, "ISugarHelper");
+
+  const Q96 = BigNumber.from(2).pow(96);
+  const sqrtRatioX96TickLower = BigNumber.from("79224201403219477170569942574"); // -1 tick
+  const sqrtRatioX96TickHigher = BigNumber.from(
+    "79228162514264337593543950336"
+  ); // 0 tick
+
+  const { tick, sqrtPriceX96 } = await pool.connect(signer).slot0({ blockTag });
+  const { liquidityGross } = await pool.connect(signer).ticks(-1, { blockTag });
+  const { amount0: tickWethBalance, amount1: tickOethBalance } =
+    await sugarHelper
+      .connect(signer)
+      .getAmountsForLiquidity(
+        sqrtPriceX96,
+        sqrtRatioX96TickLower,
+        sqrtRatioX96TickHigher,
+        liquidityGross,
+        { blockTag }
+      );
+
+  // Pool balances
+  const poolPrice = sqrtPriceX96
+    .mul(sqrtPriceX96)
+    .mul(10000000000)
+    .div(Q96)
+    .div(Q96);
+  const poolWethBalance = await weth
```

Comment:
good to know this isn't WETH liquidity in the pool, rather WETH balance/tokens of the pool. 
WETH balance = WETH liquidity + unclaimed WETH
Unclaimed WETH can originate from decreasing liquidity in the pool and not yet collecting it. (the decrease + collect are separate calls). 
In most cases though the WETH on the pool will probably be the pool's liquidity. 
Also I think the WETH swap fees remain on the pool (since those are claimed by the veAreo voters)

Reply:
I'll add the pool balances so we can see when its different to the actual WETH balance

Reply:
I thought there was a `getBalances` function but there isn't. It needs to be calculated from the sqrtPriceX96 and liquidity. I'll leave for now but can add if you thinks its useful information

Reply:
Yeah it is pretty annoying that there isn't a simple way to get that data off of the contract